### DISCUSSION TITLE
Timeout while waiting for headers

### DISCUSSION THREAD

**Discussion by josecelano** [rwf2/Rocket#2774] [Questions]
Relates to: https://github.com/hyperium/hyper/issues/3178

Can we set up a timeout for the client to send the request after opening a connection?

**NOTICE**: I don't want to set up a timeout for the client sending the headers.

I've created a sample repo that explains it better:

https://github.com/josecelano/axum-server-timeout

**Comment by SergioBenitez** [comment]
At the moment we're limited by what hyper offers here, which isn't great. But #1405 tracks implementing a feature that would resolve this.
